### PR TITLE
DEMO 327 (Feature): Create Scenario Data Display

### PR DESCRIPTION
**Pull request by Paterma**
**JIRA Ticket [Create Scenario Data Display](https://rocketcom.atlassian.net/browse/DEMO-327)**

## Brief Description
- Create Scenario Data Display (with fake values for now)
- Create popup menu for icon click next to applicable data values
- Create custom pipe to convert units when items in rux-menu are clicked 
- Create dialog screen and dialog route for settings icon

## Issues and/or Limitations
- Right now if you open the dialog, then close it, and hit back in the browser, it will take you back to that dialog route. I believe we will need to set up [Route Guards](https://www.digitalocean.com/community/tutorials/angular-route-guards) to prevent this from happening, _if_ that's something we want to prevent. I will get design's input on this and make a separate ticket if necessary. 
- There is no example for the units to be selected in the popup for inclination. So for now I took a guess and those will may need to be updated. 
- I'll be creating a sub ticket to handle state to avoid this PR getting too large

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [x] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/properties-dialog/properties-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        <rux-select>
+          <rux-option label="Drop Down" />
+        </rux-select>
+      </span>
+      <span>
+        <rux-checkbox label="Data Display" />
+        <rux-select>
+          <rux-option label="Drop Down" />
+        </rux-select>
+      </span>
```
Comment (NicDominguez) [inline src/app/properties-dialog/properties-dialog.component.html:60]: Please use a for loop and build an array of possible properties to construct it from.

File: src/app/unit-converter.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      return value * 1000;
+    } else if (unit === 'mi') {
+      return value * 0.621371;
+    } else if (unit === 'deg') {
+      return `${value}\u00B0`;
+    } else if (unit === '%') {
+      return `${value}%`;
+    } else {
+      return value;
+    }
```
Comment (NicDominguez) [inline src/app/unit-converter.pipe.ts:19]: This needs to be revisited. Each property will have a default unit type (most likely metric system). The default value would not change but the correct unit suffix would be appended. If it is not the default unit type then the correct arithmetic should be done to transform it and the new suffix appended. I was originally thinking that there would be different pipe for each conversion and they would be conditionally used. I could be wrong there and @nortonprojects may want to weigh in on what is best practice. The m/km/miles conversion work but I don't think percentage is the conversion from degrees, it is most likely radians instead.

**Review by NicDominguez — CHANGES_REQUESTED** [review]
I should made the acceptance criteria more clear but I was thinking that this should still be functional but with place holder data. So one should be able to select from a list of properties in the dialog and the dashboard list will reflect that change and vice versa. My first thought is that this state of which properties are selected for this list would be held outside of any global state so I think you should be able to do it in this PR. Let me know your thoughts if you think otherwise and we can figure it out together.

File: src/app/properties/dialog.service.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@angular/core';
```
Comment (Paterma) [inline src/app/properties/dialog.service.ts:1]: I will be using this service to handle state in a separate ticket

**Review by NicDominguez — APPROVED** [review]

**Pull request merged**